### DISCUSSION TITLE
SOLVED: Overview Span graph mistake?

### DISCUSSION THREAD

**Discussion by KryopirEinfal** [open-telemetry/opentelemetry-specification#1849] [General]
On the one hand we have this example of children spans
```
        [Span A]  ←←←(the root span)
            |
     +------+------+
     |             |
 [Span B]      [Span C] ←←←(Span C is a `child` of Span A)
     |             |
 [Span D]      +---+-------+
               |           |
           [Span E]    [Span F]
```
and the equivalent temporal relationship below:

```
––|–––––––|–––––––|–––––––|–––––––|–––––––|–––––––|–––––––|–> time

 [Span A···················································]
   [Span B··············································]
      [Span D··········································]
    [Span C········································]
         [Span E·······]        [Span F··]
```

But here "C" appears as a child of "D", not "A", I would expect it to be like this instead:

```
––|–––––––|–––––––|–––––––|–––––––|–––––––|–––––––|–––––––|–> time

 [Span A·················································································]
   [Span B··············································][Span C·········]
      [Span D··········································]
         [Span E·······]        [Span F··]
```

Does this make sense, or am I misunderstanding something?

Thanks!

**Comment by yurishkuro** [comment]
Temporal diagram does not reflect parent-child relationship. Many UIs would add a tree view to the left of the spans that would show C as a child of A.

**Reply by KryopirEinfal** [reply]
I understand, but wouldnt that then imply that Span A is  multithreaded? The only way that two children of A are concurrent is if they'd run on different threads?

What kind of UIs are you thinking of, can you point me to an example?

Thanks a lot for your prompt reply

**Reply by Oberon00** [reply]
> I understand, but wouldnt that then imply that Span A is multithreaded?

Something like that: it means that the operation traced by span A does not wait for the result of B before also starting C. B and C could run on different threads, could run on the same thread using some cooperative multithreading (e.g. Node.js event loop or other async/await mechanisms), or could run on different hosts entirely (e.g. A could send a message to a messaging system and B would process the message).

**Reply by yurishkuro** [reply]
> What kind of UIs are you thinking of, can you point me to an example?

Jaeger would render spans as shown in the first diagram.

Vertical positioning of spans does not imply anything about their concurrent execution, but their horizontal overlap (ie on the timeline) does.

**Reply by KryopirEinfal** [reply]
Got it, so the implication is indeed that 'B' 'C' are concurrent/running on diff threads, which was not what i expected from reading the doc, but thanks a lot, this clears things up.
